agent/src/main/java/io/pyroscope/javaagent/PyroscopeAgent.java

```diff
@@ -48,6 +48,21 @@ public static void start(Options options) {
         }
     }
 
+    /**
+     * stop is used to stop profiling
+     * @param options
+     */
+    public static void stop(Options options) {
```

Comment:
I think rather than passing Options here you, you should go a slightly different path.

I am proposing you replace the `AtomicBool` with an `AtomicReference<Options>`, which gets set by `start()`.

Then the stop method can stop those `<Options>` of the previous start call and set it back to null.

Reply:
That makes sense, let me give that a shot and get back to you with questions.

Reply:
I have made the changes, let me know if they make sense or if I am still missing somethin here

Reply:
@simonswine pls re-review and let me know comments